telegraf/delphix-telegraf-service

```diff
@@ -23,6 +29,13 @@ if engine_is_object_based; then
 	else
 		cat $DOSE_INPUTS $BASE_CONFIG > $TELEGRAF_CONFIG
 	fi
+elif engine_is_dct; then
+	usermod -aG docker telegraf
```

Comment:
This seems like something which should be managed when installing the package, not when it starts, so that if it fails for any reason this is caught during install / upgrade. Also, per Delphix standards, check the exit code and fail the script if this failed.

Reply:
That's a good point. The choice between the package and service time is tough here with pros and cons for each. I have switched to the package-level and had to introduce an awkward package dependency `performance-diagnostics`->`docker.io`, so when we install `performance-diagnostics` we are sure that `telegraf` user and `docker` user group exist ,so we can add the user to the user group.

Reply:
The dependency is unfortunate. If we ever ship an engine version without the need to run docker (although that does not exist today), it would be unexpected for this to start the docker service just for that.